src/ajax.js

```diff
@@ -270,7 +270,23 @@ function ajaxConvert( s, response, jqXHR, isSuccess ) {
 		}
 	}
 
-	return { state: "success", data: response };
+	if ( jQuery.isThenable(response) ) {
+		return response.then(
+			function(response) {
+				this.responseJSON = response;
+				return { state: "success", data: response };
+			}.bind(jqXHR),
+			function(e) {
+				if ( this[ "throws" ] ) {
+					throw e;
```

Comment:
Throwing here would be entirely invisible since the end user has no way to attach an error handler at the end of the Promise chain.  The only valid thing to do is have the ajax promise be rejected